### PR TITLE
Remove sympy entirely from simulator/app

### PR DESCRIPTION
**Pull request by Speculative**
This continues #11. With this change, time to simulate 100 turns is down to 30 seconds on my machine.

The flame graph is now dominated by the time to actually execute the algorithms to update the map an check for cut off units.
image

I also converted `voronoi_app` to use shapely instead of sympy. As a result of this change, `voronoi_app` is also much faster to render states (~1 second on my machine). This change was necessary because `voronoi_app` can no longer count on `voronoi_game` state containing sympy values. I tried my best to make sure the simulator still looks the same, but it's possible there are some visual changes. In this screenshot, everything is the same AFAICT except for the home bases appearing behind the Voronoi diagram for some reason
image

### CODE REVIEW

File: voronoi_app.py [hunk 1]
```diff
@@ -209,26 +213,25 @@ def reset_svgplot(self, full_refresh=False):
...
-                   sympy.geometry.Point2D(75, 75), sympy.geometry.Point2D(75, 25)]
-            q_t = sympy.geometry.Polygon([0, 0], [0, 50], [50, 50], [50, 0])
+            tmp = [Point(25, 25), Point(25, 75),
+                   Point(75, 75), Point(75, 25)]
+            q_t = [[0, 0], [0, 50], [50, 50], [50, 0], [0, 0]]
             for i in range(constants.no_of_players):
-                base_marker = sympy.geometry.Polygon(
-                    *list(map(lambda p: p.translate(tmp[i].x - q_t.centroid.x,
-                                                    tmp[i].y - q_t.centroid.y), q_t.vertices)))
+                off_x = tmp[i].x - 25
```
Comment (Speculative) [inline voronoi_app.py:220]: Logic change: inferred that the centroid is 25, 25

File: voronoi_app.py [hunk 2]
```diff
@@ -209,26 +213,25 @@ def reset_svgplot(self, full_refresh=False):
...
-            unit_t = sympy.geometry.Polygon([0, 0], [0, 1], [1, 1], [1, 0])
+            unit_t = [[0, 0], [0, 1], [1, 1], [1, 0], [0, 0]]
             for i in range(constants.max_map_dim):
                 for j in range(constants.max_map_dim):
                     if self.voronoi_game.map_states[self.curr_day][self.curr_state][i][j] != \
                             self.voronoi_game.map_states[self.prev_day][self.prev_state][i][j]:
-                        tile = sympy.geometry.Polygon(
-                            *list(map(lambda p: p.translate(i - unit_t.centroid.x + 0.5,
-                                                            j - unit_t.centroid.y + 0.5), unit_t.vertices)))
+                        tile = Polygon([(x + i, y + j) for x, y in unit_t])
```
Comment (Speculative) [inline voronoi_app.py:234]: Logic change: inferred that the centroid is 0.5, 0.5
TODO: I think this should actually be `x + i + 1` and `y + j + 1`?

File: voronoi_game.py [hunk 3]
```diff
@@ -376,36 +377,37 @@ def check_move(self, move):
             return False
         is_valid = False
         if isiterable(move) and count_iterable(move) == 2:
-            if np.all([x is not None and sympy.simplify(x).is_real for x in move]):
+            if all(x is not None and not np.isnan(x) and np.isfinite(x) for x in move):
```
Comment (Speculative) [inline voronoi_game.py:380]: Logic change: I don't know what simplifying and checking `is_real` does here, I assumed it was just a check that the move is composed of valid finite numbers

File: voronoi_game.py [hunk 4]
```diff
@@ -376,36 +377,37 @@ def check_move(self, move):
...
+        angle = float(angle)
+        a = int(self.unit_pos[day][0][idx][unit_idx].x)
+        b = int(self.unit_pos[day][0][idx][unit_idx].y)
         if distance > 1.0:
             distance = 1.0
             self.logger.debug("Distance rectified to max distance of 1 km")
 
-        new_a = a + (distance * sympy.cos(angle))
-        new_b = b + (distance * sympy.sin(angle))
+        new_a = a + (distance * np.cos(angle))
```
Comment (Speculative) [inline voronoi_game.py:393]: Replaced all calls to sympy math functions with the equivalent numpy functions.

File: voronoi_app.py (hunk 1 above)
Comment (Shreeyak) [reply voronoi_app.py:220]: Is that for any map size?

**Pull request merged**